pulsar-common/src/main/java/org/apache/pulsar/common/naming/Metadata.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.naming;
+
+import java.util.Map;
+
+public class Metadata {
+
+    private static final int MAX_METADATA_SIZE = 1024; // 1 Kb
+
+    private Metadata() {}
+
+    public static boolean validateMetadata(Map<String, String> metadata) {
+        if (metadata == null) {
+            return true;
+        }
+
+        int size = 0;
+        for (Map.Entry<String, String> e : metadata.entrySet()) {
+            size += (e.getKey().length() + e.getValue().length());
+            if (size > MAX_METADATA_SIZE) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public static String getErrorMessage() {
```

Comment:
I placed it in here i so it could be reused for the consumer pr that is coming. Also, if the size increases there is only one place we need to update the code? thoughts?

Reply:
sure. then should we keep error-string as public variables in case `Metadata` validation has multiple exception scenarios?

Reply:
we could have the validate method return an error string on error, otherwise null if everything is ok. thoughts?

Reply:
Instead of having the `boolean` return type and the error message, we could instead make the `validateMetadata()` to throw exception when it's not valid